### DISCUSSION TITLE
Multi-GPU Sharding

### DISCUSSION THREAD

**Discussion by gorlando04** [facebookresearch/faiss#3562] [General]
# Summary

Faiss version: Faiss 1.6.3
Installed from: anaconda (pytorch)

Running on:
- [ ] CPU
- [ X] GPU

Interface: 
- [ ] C++
- [ X] Python

# Reproduction instructions

Hi, I'm trying to do a multi-gpu kNN search using faiss. But i'm having problem in working with sharding. Replication is an easy solution but this approach limits the datasets size, so i want to understand how i could implement a IVFPQ index, using Multi-GPU sharding.

**Comment by mdouze** [comment]
see 
https://github.com/facebookresearch/faiss/wiki/Faiss-on-the-GPU#using-multiple-gpus

**Comment by gorlando04** [comment]
Hello @mdouze, thank you very much for the help, with this i could implement a GPU sharding script, with the additional help of the bench_gpu_1bn.py script.

Although, I have one final question. Is sharding supposed to be slower than Replication? In my tests sharding was slower than replication for datasets with smaller sizes like 10M - 50M, is this normal? If yes, is there an explanation why
